src/RetrySettings.php

```diff
@@ -217,6 +217,13 @@ class RetrySettings
 
     private $noRetriesRpcTimeoutMillis;
 
+    /**
+     * When set, this function will be used to evaluate if the retry should
+     * take place or not. The callable will have the following signature:
+     * function (Exception $e, array $options, int $retryAttempt): bool
+     */
+    private $requestRetryFunction;
```

Comment:
nit: rename all these to `retryFunction`
```suggestion
    private $retryFunction;
```